### PR TITLE
Misc remove `&x` ccall  syntax

### PR DESCRIPTION
**Pull request by musm**

### CODE REVIEW

File: base/irrationals.jl [hunk 1]
```diff
@@ -119,10 +119,10 @@ macro irrational(sym, val, def)
     qsym = esc(Expr(:quote, sym))
     bigconvert = isa(def,Symbol) ? quote
         function Base.convert(::Type{BigFloat}, ::Irrational{$qsym})
-            c = BigFloat()
+            c = Ref(BigFloat())
             ccall(($(string("mpfr_const_", def)), :libmpfr),
                   Cint, (Ref{BigFloat}, Int32), c, MPFR.ROUNDING_MODE[])
-            return c
+            return c[]
```
Comment (yuyichao) [inline base/irrationals.jl]: Why?

Comment (musm) [inline base/irrationals.jl]: I am following the suggestion here https://github.com/JuliaMath/SpecialFunctions.jl/pull/44#discussion_r133771693

Comment (yuyichao) [inline base/irrationals.jl]: The Ref is not doing anything and you **ARE** mutating the `BigFloat` so adding a `Ref` seems like a bad idea.

Comment (musm) [inline base/irrationals.jl]: OK thank you very much for clarifying this! Droped the last commit. Other than that is this LGTY?

File: base/printf.jl [hunk 2]
```diff
@@ -1130,8 +1130,8 @@ function bigfloat_printf(out, d, flags::String, width::Int, precision::Int, c::C
     @assert length(printf_fmt) == fmt_len
     bufsiz = length(DIGITS)
     lng = ccall((:mpfr_snprintf,:libmpfr), Int32,
-                (Ptr{UInt8}, Culong, Ptr{UInt8}, Ptr{BigFloat}...),
-                DIGITS, bufsiz, printf_fmt, &d)
+                (Ptr{UInt8}, Culong, Ptr{UInt8}, Ref{BigFloat}),
```
Comment (yuyichao) [inline base/printf.jl]: Keep the `...` It can affect calling convention (and does on x86_64).

**Review by yuyichao — CHANGES_REQUESTED** [review]

**Review by yuyichao — APPROVED** [review]

**Comment by musm** [comment]
Thanks for the review @yuyichao

**Comment by musm** [comment]
part of https://github.com/JuliaLang/julia/issues/6080 (ref)

**Comment by musm** [comment]
good to merge?

**Review by rfourquet — APPROVED** [review]

**Pull request merged**